evennia/server/inputfuncs.py

```diff
@@ -654,6 +662,119 @@ def msdp_send(session, *args, **kwargs):
     session.msg(send=((), out))
 
 
+def is_typing_send_update():
+    """
+    Send relevant updates to participants
+
+    """
+    participants = list(_IS_TYPING_PARTICIPANTS.keys())
+
+    for participant in participants:
+        if _IS_TYPING_PARTICIPANTS[participant]["session"].puppet is not None:
+
+            payload = []
+            # Get potentials
+            potentials = (DefaultCharacter.objects.filter_family(db_location=_IS_TYPING_PARTICIPANTS[participant]["session"].puppet.location)
+                          .exclude(db_key=_IS_TYPING_PARTICIPANTS[participant]["session"].puppet.key))
+
+            for puppet in potentials:
+
+                # We're only interested in sending updates if they're capable of receiving them
+                if str(puppet.sessid) in participants:
+                    payload.append({
+                        "name": puppet.name,
+                        "state": _IS_TYPING_PARTICIPANTS[str(puppet.sessid)]['state'],
+                    })
+
+            _IS_TYPING_PARTICIPANTS[participant]['session'].msg(is_typing={
+                'type': 'typing',
+                'payload': payload
+            })
+            delay(5, is_typing_send_update)
```

Comment:
The delays may be what is causing the server issues, but it's not clear why. Maybe it would be worth experimenting with a direct send here (no delay) just to see if that resolves things.

Reply:
That's how it worked originally--no issues. I have rewritten this four times from when I had it working without the server batching the responses. This issue isn't directly with delay(), it's with TICKER_HANDLER. If I so much as looked at it directly, the server would be broken on restart to the point where I'd have to nuke the install and I wasn't saving anything to the DB so nothing should have changed... It's especially interesting in that using delay can prevent the server from restarting. Regardless, nothing is logged anywhere that I could...

I'll roll the whole thing back to just relaying in real time, clean  it up, separate it, and see about writing tests.